system/Queue/Handlers/DatabaseHandler.php

```diff
@@ -0,0 +1,180 @@
+<?php namespace CodeIgniter\Queue\Handlers;
+
+/**
+ * Queue handler for database.
+ */
+class DatabaseHandler implements QueueHandlerInterface
+{
+	const STATUS_WAITING   = 10;
+	const STATUS_EXECUTING = 20;
+	const STATUS_DONE      = 30;
+	const STATUS_FAILED    = 40;
+	protected $group_config;
+	protected $config;
+	protected $db;
+
+	/**
+	 * constructor.
+	 *
+	 * @param  array $group_config
+	 * @param  \Codeigniter\Config\Queue $config
+	 */
+	public function __construct($group_config, \Codeigniter\Config\Queue $config)
+	{
+		$this->group_config = $group_config;
+		$this->config       = clone $config;
+		$this->db           = \Config\Database::connect($this->group_config['dbGroup'], $this->group_config['sharedConnection']);
+	}
+
+	/**
+	 * Setup queueing system (system-wide).
+	 */
+	/*
+	public function setup()
+	{
+		CREATE TABLE ci_queue (
+			id          int(11) NOT NULL AUTO_INCREMENT,
+			queue_name  varchar(255) NOT NULL,
+			status      int(11) NOT NULL,
+			weight      int(11) NOT NULL,
+			retry_count int(11) NOT NULL DEFAULT 0,
+			exec_after  datetime NOT NULL,
```

Comment:
I see you have a exec_after command. This means we'd need a way to set that, like a `delay()` method or similar?